README.md

```diff
@@ -1,6 +1,64 @@
 # dhall-bhat
 Tasty meal of Dhall
 
+We currently require Dhall 1.18.0, and try to support newer releases as much as possible.
+
 ## Documentation
 
 Types as documentation: https://formationai.github.io/dhall-bhat/
+
+## Style
+
+Dhall is still a very young community, so the style is still evolving. Here’s how we’re currently doing things in dhall-bhat.
+
+### group expressions by type
+
+It’s common to define a type and a number of related expressions for it. When you do this, we generally create a directory named for the type, with everything related to it in that directory.
+
+- Foo (the name of the type, a directory)
+  - Type (the file containing the type itself, capitalized)
+  - functor (file contaning a Functor instance for this type, lowercase)
+  - op (some operation defined on this type, lowercase)
+  - package.dhall (file contanining a record of all the terms related to this type)
+
+### use `dhall format`
+
+`make format` (or, even better, `make lint`) in this repo will automatically apply this to the entire repository.
+
+**NB**: `dhall format` (and `lint`) currently [removes all comments from the file](https://github.com/dhall-lang/dhall-haskell/issues/145) except for a single “heading” comment. So be careful that you don’t lose important comments this way.
```

Comment:
Ah, just read this now. Should probably note that unused expressions will be removed as well if there's any WIP.